### PR TITLE
Support .NET 9

### PR DESCRIPTION
**Pull request by WhitWaldo**
# Description

I've updated the build and integration test scripts to include support for the latest patch release of .NET 9 (which I believe will capture RC builds, but I couldn't find much information on it outside of the `actions/setup-dotnet` GitHub action.

I've made the necessary changes to the various projects in the solution to update references for each of the now-supported .NET framework versions: .NET 6, .NET 8 and .NET 9 (and .NET 7 with regards to Dapr.Workflows).

All the .NET SDK documentation section introduction pages have been updated to reflect the .NET 9 support, but I also added a note to make it clear that in Dapr's v1.16 release, we will be dropping support for .NET 6 and .NET 7 throughout and only supporting .NET 8 and .NET 9.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1400

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

**Review by mikeee — DISMISSED** [review]
It looks like there is a missing targetframework for .NET 9 for the E2E tests

**Comment by mikeee** [comment]
https://github.com/search?q=repo%3Adapr%2Fdotnet-sdk%20net7&type=code

Some of the locations where the target frameworks where version 9 needs to be added to

File: .github/workflows/sdk_build.yml [hunk 1]
```diff
@@ -42,39 +42,37 @@ jobs:
...
           install-version: '7.0.x'
         - dotnet-version: '8.0'
-          install-3: false
           display-name: '.NET 8.0'
           framework: 'net8'
           prefix: 'net8'
           install-version: '8.0.x'
+        - dotnet-version: '9.0'
+          display-name: '.NET 9.0'
+          framework: 'net9.0'
```
[the remark this answers is not in the document]
Comment (WhitWaldo) [reply .github/workflows/sdk_build.yml:64]: Just trying to get the thing to build though and through. When I used 'net9', it failed and a search on Bing suggested that the TFM had possibly changed for .NET 9, so I tried mixing this up. It didn't work either, so I'll revert in a future iteration.

File: daprdocs/content/en/dotnet-sdk-docs/_index.md [hunk 2]
```diff
@@ -18,7 +18,15 @@ Dapr offers a variety of packages to help with the development of .NET applicati
 
 - [Dapr CLI]({{< ref install-dapr-cli.md >}}) installed
 - Initialized [Dapr environment]({{< ref install-dapr-selfhost.md >}})
-- [.NET 6](https://dotnet.microsoft.com/download) or [.NET 8+](https://dotnet.microsoft.com/download) installed
+- [.NET 6](https://dotnet.microsoft.com/download), [.NET 8](https://dotnet.microsoft.com/download) or [.NET 9](https://dotnet.microsoft.com/download) installed
+
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
+and .NET 7 is the minimally supported version of .NET by Dapr.Workflows in Dapr v1.15, only .NET 8 and .NET 9 wil
```
Comment (philliphoff) [inline daprdocs/content/en/dotnet-sdk-docs/_index.md:26]: "will"

File: daprdocs/content/en/dotnet-sdk-docs/dotnet-actors/dotnet-actors-howto.md [hunk 3]
```diff
@@ -45,7 +45,15 @@ This project contains the implementation of the actor client which calls MyActor
 
 - [Dapr CLI]({{< ref install-dapr-cli.md >}}) installed.
 - Initialized [Dapr environment]({{< ref install-dapr-selfhost.md >}}).
-- [.NET 6+](https://dotnet.microsoft.com/download) installed. Dapr .NET SDK uses [ASP.NET Core](https://docs.microsoft.com/aspnet/core/introduction-to-aspnet-core?view=aspnetcore-6.0).
+- [.NET 6](https://dotnet.microsoft.com/download), [.NET 8](https://dotnet.microsoft.com/download) or [.NET 9](https://dotnet.microsoft.com/download) installed
+
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
+and .NET 7 is the minimally supported version of .NET by Dapr.Workflows in Dapr v1.15, only .NET 8 and .NET 9 wil
```
Comment (philliphoff) [inline daprdocs/content/en/dotnet-sdk-docs/dotnet-actors/dotnet-actors-howto.md:53]: "will"

File: daprdocs/content/en/dotnet-sdk-docs/dotnet-jobs/dotnet-jobs-howto.md [hunk 4]
```diff
@@ -16,10 +16,17 @@ In the .NET example project:
...
-- [.NET 6+](https://dotnet.microsoft.com/download) installed
 - [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/)
 - [Initialized Dapr environment](https://docs.dapr.io/getting-started/install-dapr-selfhost)
-- [Dapr Jobs .NET SDK](https://github.com/dapr/dotnet-sdk)
+- [.NET 6](https://dotnet.microsoft.com/download), [.NET 8](https://dotnet.microsoft.com/download) or [.NET 9](https://dotnet.microsoft.com/download) installed
+
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
+and .NET 7 is the minimally supported version of .NET by Dapr.Workflows in Dapr v1.15, only .NET 8 and .NET 9 wil
```
Comment (philliphoff) [inline daprdocs/content/en/dotnet-sdk-docs/dotnet-jobs/dotnet-jobs-howto.md:26]: "will"

File: daprdocs/content/en/dotnet-sdk-docs/dotnet-workflow/dotnet-workflow-howto.md [hunk 5]
```diff
@@ -18,11 +18,17 @@ In the .NET example project:
...
-- [.NET 6+](https://dotnet.microsoft.com/download) installed
 - [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/)
 - [Initialized Dapr environment](https://docs.dapr.io/getting-started/install-dapr-selfhost/)
-- [Dapr .NET SDK](https://github.com/dapr/dotnet-sdk/)
+- [.NET 7](https://dotnet.microsoft.com/download), [.NET 8](https://dotnet.microsoft.com/download) or [.NET 9](https://dotnet.microsoft.com/download) installed
 
+{{% alert title="Note" color="primary" %}}
+
+Note that while .NET 6 is generally supported as the minimum .NET requirement across the Dapr .NET SDK packages
+and .NET 7 is the minimally supported version of .NET by Dapr.Workflows in Dapr v1.15, only .NET 8 and .NET 9 wil
```
Comment (philliphoff) [inline daprdocs/content/en/dotnet-sdk-docs/dotnet-workflow/dotnet-workflow-howto.md:28]: "will"

File: global.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "_comment": "This policy allows the 8.0.100 SDK or patches in that family.",
   "sdk": {
-    "version": "8.0.100",
-    "rollForward": "minor"
+    "version": "9.0.100-rc.2",
```
Comment (philliphoff) [inline global.json:4]: Should be able to drop the "rc.2" now?

File: src/Dapr.Jobs/Models/DaprJobSchedule.cs [hunk 7]
```diff
@@ -67,7 +67,9 @@ public static DaprJobSchedule FromCronExpression(CronExpressionBuilder builder)
     /// <returns></returns>
     public static DaprJobSchedule FromDateTime(DateTimeOffset scheduledTime)
     {
+#if NET6_0
         ArgumentNullException.ThrowIfNull(scheduledTime, nameof(scheduledTime));
```
Comment (philliphoff) [inline src/Dapr.Jobs/Models/DaprJobSchedule.cs:71]: Can a `DateTimeOffset` (a `struct`) ever be null?

File: daprdocs/content/en/dotnet-sdk-docs/_index.md (hunk 2 above)
Comment (WhitWaldo) [reply daprdocs/content/en/dotnet-sdk-docs/_index.md:26]: Good catch - fixed that here and in the other PR I copy/pasted this into.

File: src/Dapr.Jobs/Models/DaprJobSchedule.cs (hunk 7 above)
Comment (WhitWaldo) [reply src/Dapr.Jobs/Models/DaprJobSchedule.cs:71]: Haha, bad habit.

**Review by philliphoff — APPROVED** [review]

**Pull request merged**